src/draw/lv_image_decoder.c

```diff
@@ -321,6 +299,70 @@ lv_draw_buf_t * lv_image_decoder_post_process(lv_image_decoder_dsc_t * dsc, lv_d
     return decoded;
 }
 
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+static lv_image_decoder_t * image_decoder_get_info(const void * src, lv_image_header_t * header)
+{
+    lv_memzero(header, sizeof(lv_image_header_t));
+
+    if(src == NULL) return NULL;
+
+    lv_image_src_t src_type = lv_image_src_get_type(src);
+    if(src_type == LV_IMAGE_SRC_VARIABLE) {
+        const lv_image_dsc_t * img_dsc = src;
+        if(img_dsc->data == NULL) return NULL;
+    }
+
+    lv_image_decoder_t * decoder;
+
+#if LV_IMAGE_HEADER_CACHE_DEF_CNT > 0
+    lv_image_header_cache_data_t search_key;
+    search_key.src_type = src_type;
+    search_key.src = src;
+
+    lv_cache_entry_t * entry = lv_cache_acquire(img_header_cache_p, &search_key, NULL);
+
+    if(entry) {
+        lv_image_header_cache_data_t * cached_data = lv_cache_entry_get_data(entry);
+        *header = cached_data->header;
+        decoder = cached_data->decoder;
+        lv_cache_release(img_header_cache_p, entry, NULL);
+        return decoder;
+    }
+#endif
+
+    _LV_LL_READ(img_decoder_ll_p, decoder) {
+        /*Info and Open callbacks are required*/
+        if(decoder->info_cb && decoder->open_cb) {
+            lv_result_t res = decoder->info_cb(decoder, src, header);
+            if(res == LV_RESULT_OK) {
+                if(header->stride == 0) header->stride = img_width_to_stride(header);
+                break;
+            }
+        }
+    }
+
+#if LV_IMAGE_HEADER_CACHE_DEF_CNT > 0
+    if(decoder) {
+        if(src_type == LV_IMAGE_SRC_FILE) search_key.src = lv_strdup(src);
+        search_key.decoder = decoder;
+        search_key.header = *header;
+        entry = lv_cache_add(img_header_cache_p, &search_key, NULL);
+
+        if(entry == NULL) {
+            if(src_type == LV_IMAGE_SRC_FILE) lv_free((void *)src);
```

Comment:
I think the duplicated string should be freed (otherwise there is a memory leak):
`lv_free(search_key.src)`

Reply:
You're right, my mistakes. 😭

Reply:
Done